clients/src/main/java/org/apache/kafka/common/requests/OffsetCommitResponse.java

```diff
@@ -190,12 +203,85 @@ public Builder merge(
                     }
                 });
             }
-
             return this;
         }
 
         public OffsetCommitResponse build() {
             return new OffsetCommitResponse(data);
         }
     }
+
+    public static class TopicIdBuilder extends Builder {
+        private final HashMap<Uuid, OffsetCommitResponseTopic> byTopicId = new HashMap<>();
+
+        @Override
+        protected void add(OffsetCommitResponseTopic topic) {
+            throwIfTopicIdIsNull(topic.topicId());
+            data.topics().add(topic);
+            byTopicId.put(topic.topicId(), topic);
+        }
+
+        @Override
+        protected OffsetCommitResponseTopic get(Uuid topicId, String topicName) {
+            throwIfTopicIdIsNull(topicId);
+            return byTopicId.get(topicId);
+        }
+
+        @Override
+        protected OffsetCommitResponseTopic getOrCreate(Uuid topicId, String topicName) {
+            throwIfTopicIdIsNull(topicId);
+            OffsetCommitResponseTopic topic = byTopicId.get(topicId);
+            if (topic == null) {
+                topic = new OffsetCommitResponseTopic()
+                    .setName(topicName)
+                    .setTopicId(topicId);
+                data.topics().add(topic);
+                byTopicId.put(topicId, topic);
+            }
+            return topic;
+        }
+
+        private static void throwIfTopicIdIsNull(Uuid topicId) {
```

Comment:
Not sure whether `throwIfTopicIdIsNull` and `throwIfTopicNameIsNull` is redundant. When we build `OffsetCommitResponseTopic`, the `topicId` and `topicName` data are from `OffsetCommitRequestTopic`. In the generated code, the default name is `""` and default `topicId` is `Uuid.ZERO_UUID`.

Reply:
You're right. It may not be needed but being defensive does not hurt, I guess.